tests/main/main_test.go

```diff
@@ -0,0 +1,9 @@
+package main
+
+import "testing"
+
+func TestMain(t *testing.T) {
```

Comment:
@neelance, are you aware that this name `TestMain` has special meaning as of Go 1.4?

> The `testing` package has a new facility to provide more control over running a set of tests. If the test code contains a function
> 
> ```
> func TestMain(m *testing.M) 
> ```
>
> that function will be called instead of running the tests directly. The M struct contains methods to access and run the tests.

Source: https://golang.org/doc/go1.4

I kinda suspect it's an unintentional usage of that feature here, since you're testing a feature named main. If so, I think it's better we rename it to avoid doing something unintentionally just due to a coincidence in naming, as it might cause confusion later (if other tests are ever added here, they will not run). Perhaps `func TestNotRunMain()` or so?